src/app/pages/signup-page/signup-page.component.html

```diff
@@ -1 +1,68 @@
-<p>signup-page works!</p>
+<h2>SighUp</h2>
+
+<form [formGroup]="registryForm" (ngSubmit)="onSubmit()" [ngClass]="firstContact ? 'active' : null">
+  <mat-form-field appearance="outline">
+    <mat-label for="email">Email</mat-label>
+    <input matInput type="text" id="email" formControlName="email" (keydown)="onClearError()" />
+    <mat-error *ngIf="email.hasError('required') && email.dirty && email.touched && firstContact">
```

Comment:
maybe you don't need all of this checkers, watch example with email https://material.angular.io/components/form-field/examples  and implement it in your component. Also did it for password field too. 
If you need help with this task, I can send you my login form screenshots from youtube project